cluster_experiments/inference/hypothesis_test.py

```diff
@@ -0,0 +1,196 @@
+import copy
+from typing import List, Optional
+
+import pandas as pd
+
+from cluster_experiments.cupac import CupacHandler
+from cluster_experiments.experiment_analysis import InferenceResults
+from cluster_experiments.inference.dimension import DefaultDimension, Dimension
+from cluster_experiments.inference.metric import Metric
+from cluster_experiments.inference.variant import Variant
+from cluster_experiments.power_config import analysis_mapping
+
+
+class HypothesisTest:
+    """
+    A class used to represent a Hypothesis Test with a metric, analysis, optional analysis configuration, and optional dimensions.
+
+    Attributes
+    ----------
+    metric : Metric
+        An instance of the Metric class
+    analysis : ExperimentAnalysis
+        An instance of the ExperimentAnalysis class
+    analysis_config : Optional[dict]
+        An optional dictionary representing the configuration for the analysis
+    dimensions : Optional[List[Dimension]]
+        An optional list of Dimension instances
+    cupac_config : Optional[dict]
+            An optional dictionary representing the configuration for the cupac model
+    """
+
+    def __init__(
+        self,
+        metric: Metric,
+        analysis_type: str,
+        analysis_config: Optional[dict] = None,
+        dimensions: Optional[List[Dimension]] = None,
+        cupac_config: Optional[dict] = None,
+    ):
+        """
+        Parameters
+        ----------
+        metric : Metric
+            An instance of the Metric class
+        analysis_type : str
+            string mapper to an ExperimentAnalysis
+        analysis_config : Optional[dict]
+            An optional dictionary representing the configuration for the analysis
+        dimensions : Optional[List[Dimension]]
+            An optional list of Dimension instances
+        cupac_config : Optional[dict]
+            An optional dictionary representing the configuration for the cupac model
+        """
+        self._validate_inputs(metric, analysis_type, analysis_config, dimensions)
+        self.metric = metric
+        self.analysis_type = analysis_type
+        self.analysis_config = analysis_config or {}
+        self.dimensions = [DefaultDimension()] + (dimensions or [])
```

Comment:
oh, but this is used in the analysis_plan code, because in the most generic case each test can have a different list of dimensions to slice on. This is a good question, not sure if there's a better design